soot-infoflow-android/src/soot/jimple/infoflow/android/data/AndroidMethod.java

```diff
@@ -164,6 +164,11 @@ public boolean equals(Object obj) {
 			return false;
 		if (sourceSinkType != other.sourceSinkType)
 			return false;
+		if (!this.className.equals(other.className)
```

Comment:
That should be handled on the superclass ``SootMethodAndClass`` and we have a call to ``super.equals`` here. Can you please check why this is not sufficient?